packages/starlight-llms-txt/entryToSimpleMarkdown.ts

```diff
@@ -83,7 +83,27 @@ const htmlToMarkdownPipeline = unified()
 				// This is what `hast-util-to-mdast` checks for code language metadata.
 				if (pre?.properties.dataLanguage && code) {
 					if (!Array.isArray(code.properties.className)) code.properties.className = [];
-					code.properties.className.push(`language-${pre.properties.dataLanguage}`);
+
+					const diffLines =
+						pre.properties.dataLanguage === 'diff'
+							? []
+							: code.children.filter((child) => matches('div.ec-line.ins, div.ec-line.del', child));
+					if (diffLines.length === 0) {
+						code.properties.className.push(`language-${pre.properties.dataLanguage}`);
+					} else {
+						code.properties.className.push('language-diff');
+						for (const line of diffLines) {
+							if (line.type !== 'element') continue;
+							const classes = line.properties?.className;
+							if (typeof classes !== 'string' && !Array.isArray(classes)) continue;
+							const marker = classes.includes('ins') ? '+' : '-';
+							const span = select('span:not(.indent)', line);
+							const firstChild = span?.children[0];
+							if (firstChild?.type === 'text') {
+								firstChild.value = `${marker}${firstChild.value}`;
+							}
+						}
```

Comment:
@hippotastic does this seem OK to you as a way to reliably update `diff` code blocks? Any EC conditions it wouldn’t meet?

Reply:
This looks like a good start! I'd probably remove the check `pre.properties.dataLanguage === 'diff'` because EC allows adding del/ins markers to blocks not using the `diff` language as well.

Also, this code covers full line changes only if I understand it correctly. It might be a good idea to check for inline `<del>` and `<ins>` tags inside code blocks as well and remove the contents of `<del>` ones.

Reply:
...or am I completely misunderstanding what this code is doing? :) Is it converting non-diff code blocks to diffs?

Reply:
Yes exactly this is what's it's doing. If a codeblock is a diff with a lang or a lang codeblock with ins/del markers, it transforms it to just a diff

Reply:
I don't feel like doing this now bc I'm sick but I'll open an issue so we don't forget

Reply:
#23

Reply:
I think that's fine. Progressive enhancement step by step! This version should already be better than the current one as-is.

Reply:
Thanks for taking a look!